src/lambda/expression.rs

```diff
@@ -106,3 +107,101 @@ impl Eval for Expression {
     })
   }
 }
+
+impl Expression {
+  pub(crate) fn io_iter(&mut self) -> IOExpressionIterator {
```

Comment:
special iterator that will walk through all levels of expression and return only IO operation. 
Made it only for io and not general just to solve the problem inside app_context since later we may consider some other options to work with that

Reply:
Can you add some Rust Docs on why this exists and where should it be used. Ideally we would like to make sure its not used inside of expression evaluation.

Reply:
I didn't like that it was specific for io resolvers and I've rewrote this to another method that could be considered general for working with expressions (traversing over expression tree) and just use that method in app_context for specific case